Fails to build on Windows with GHC 9.6.1

I attempted to add GHC 9.6.1 to our CI config, but the build of 6791e77 fails on Windows 2022 with:

```
[51 of 56] Compiling Hakyll.Web.Feed  ( lib\Hakyll\Web\Feed.hs, D:\\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Hakyll\Web\Feed.o )
ghc-9.6.1.exe:  | C:\sr\ghc-9.6.1\network-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109\lib\libHSnetwork-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109.a: unknown symbol `__mingw_vsprintf'
ghc-9.6.1.exe:  | C:\sr\ghc-9.6.1\network-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109\lib\libHSnetwork-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109.a: unknown symbol `getWSErrorDescr'
ghc-9.6.1.exe:  | C:\sr\ghc-9.6.1\network-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109\lib\libHSnetwork-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109.a: unknown symbol `networkzm3zi1zi2zi8zm8ccbc4dc50a14d19b54bceabfbac9ee74837a109_NetworkziSocketziInternal_throwSocketError1_info'
ghc-9.6.1.exe:  | C:\sr\ghc-9.6.1\network-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109\lib\libHSnetwork-3.1.2.8-8ccbc4dc50a14d19b54bceabfbac9ee74837a109.a: unknown symbol `networkzm3zi1zi2zi8zm8ccbc4dc50a14d19b54bceabfbac9ee74837a109_NetworkziSocketziSyscall_zdwsocket_closure'
ghc-9.6.1.exe:  | C:\sr\ghc-9.6.1\wai-3.2.3-59fda19bdf0e39e8d6b4999a23c46ed371d20a5d\lib\libHSwai-3.2.3-59fda19bdf0e39e8d6b4999a23c46ed371d20a5d.a: unknown symbol `networkzm3zi1zi2zi8zm8ccbc4dc50a14d19b54bceabfbac9ee74837a109_NetworkziSocketziInfo_zdwzdcshowsPrec2_closure'
ghc-9.6.1.exe:  | C:\sr\ghc-9.6.1\wai-app-static-3.1.7.4-a363cbf89679d48a971b5fbeeb955a12780ebf4b\lib\libHSwai-app-static-3.1.7.4-a363cbf89679d48a971b5fbeeb955a12780ebf4b.a: unknown symbol `waizm3zi2zi3zm59fda19bdf0e39e8d6b4999a23c46ed371d20a5d_NetworkziWaiziInternal_zdtcResponseReceived_closure'
ghc-9.6.1.exe:  | D:\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Hakyll\Core\Configuration.o: unknown symbol `waizmappzmstaticzm3zi1zi7zi4zma363cbf89679d48a971b5fbeeb955a12780ebf4b_WaiAppStaticziTypes_zdtcStaticSettings_closure'
ghc-9.6.1.exe: Could not load Object Code D:\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Hakyll\Core\Configuration.o.


<no location info>: error: 
[52 of 56] Compiling Paths_hakyll     ( D:\\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\autogen\Paths_hakyll.hs, D:\\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Paths_hakyll.o )
[53 of 56] Compiling Hakyll.Check     ( lib\Hakyll\Check.hs, D:\\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Hakyll\Check.o )
[54 of 56] Compiling Hakyll.Commands  ( lib\Hakyll\Commands.hs, D:\\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Hakyll\Commands.o )
[55 of 56] Compiling Hakyll.Main      ( lib\Hakyll\Main.hs, D:\\a\hakyll\hakyll\dist-newstyle\build\x86_64-windows\ghc-9.6.1\hakyll-4.15.1.1\build\Hakyll\Main.o )
Error: cabal.exe: Failed to build hakyll-4.15.1.1 (which is required by
test:hakyll-tests from hakyll-4.15.1.1 and exe:hakyll-init from
hakyll-4.15.1.1).

Error: Process completed with exit code 1.
```

There is a report against the `network` package https://github.com/haskell/network/issues/544, but it was closed without any satisfactory resolution. It's possible that GitHub Actions is the one at fault, but I'm not sure. FWIW, the build passes on Ubuntu 22.04.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -29,9 +29,6 @@ jobs:
           # fails to build: "can't load framework: Cocoa (not found)"
           - os: macOS-12
             ghc: "8.8.4"
-          # fails to link: https://github.com/jaspervdj/hakyll/issues/986
-          - os: windows-2022
-            ghc: "9.6.1"
 
     steps:
     - uses: actions/checkout@v3

